modules/renter/download.go

```diff
@@ -266,11 +253,23 @@ func (r *Renter) Download(nickname, destination string) error {
 	}
 
 	// Check that this host set is sufficient to download the file.
-	err = checkHosts(hosts, file.erasureCode.MinPieces(), file.numChunks())
+	err := checkHosts(hosts, file.erasureCode.MinPieces(), file.numChunks())
 	if err != nil {
 		return err
 	}
 
+	// Create file on disk with the correct permissions.
+	perm := os.FileMode(file.mode)
+	if perm == 0 {
+		// sane default
+		perm = 0666
+	}
+	f, err := os.OpenFile(destination, os.O_CREATE|os.O_RDWR|os.O_TRUNC, perm)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
```

Comment:
I'm not sure how SUID and SGID stuff works, but that might be a concern. I think the system is safer than that though - I don't think you'd be allowed to download a SUID program that lets anyone run as root if you are in userspace.

But I'm not sure if it would cause errors during the download, we should probably test it somehow.